src/Options.php

```diff
@@ -937,6 +951,24 @@ private function normalizeDsnOption(SymfonyOptions $options, $value): ?Dsn
         return Dsn::createFromString($value);
     }
 
+    /**
+     * Normalizes the value of the "context_lines" option and throws a deprecation
+     * warning to favor the integration instead.
+     *
+     * @param SymfonyOptions $options The options
+     * @param int|null       $value   The actual value of the option
+     *
+     * @deprecated since version 2.4, to be removed in 3.0
+     */
+    private function normalizeContextLinesOption(SymfonyOptions $options, ?int $value): ?int
+    {
+        if (null !== $value) {
+            @trigger_error('The "context_lines" option is deprecated since version 2.4 and will be removed in 3.0. Use the "Sentry\Integration\FrameContextifierIntegration" integration instead.', E_USER_DEPRECATED);
+        }
+
+        return $value;
+    }
```

Comment:
Dropping the options doesn't seem sensible to me. It would be a lot more difficult to change this default value.

Reply:
I knew we would have discussed about this :grimacing: There are quite a few reasons I want to deprecate this option:

- It's one less option to hande around the codebase, to validate, normalize, etc
- Since the feature is now controlled with an integration, it doesn't make sense at all to have a SDK option that affects a specific integration
- By adding or removing the integration you control whether you want to send the context lines or not, therefore making the special handling of a `null` value useless and this in turn makes the feature clearer about how it works (if something isn't enabled then obviously it does nothing)
- The Go SDK does not have such option as well, and this implementation is just a porting of the way it's handled there here

Reply:
I would agree on this, but we have one drawback: this should be a default enabled integration. So you would have to re-add it to make it work? How do we handle adding an integration twice?

This does not seem smooth to me..

Reply:
>I would agree on this, but we have one drawback: this should be a default enabled integration

It is at the time of writing, in the future we may play safer and not send by default the context lines as there may be some organizations not willing to share their source code with external applications, but this is out of topic here of course

>How do we handle adding an integration twice?

We shouldn't anyway handle it probably as there may be some legitimate case for having the same integration added more than once per client (I'm not talking specifically about default integrations)

Reply:
Still, it would become really complicated just to switch this off. I'm still :-1: on removing the option.

Reply:
As @HazAT said that for the moment I should refrain from deprecating the option, I reverted the changes :confused: